dali/pipeline/executor/executor2/exec_graph_test.cc

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include "dali/pipeline/executor/executor2/exec2_ops_for_test.h"
+#include "dali/pipeline/operator/arg_helper.h"
+#include "dali/pipeline/executor/executor2/exec_graph.h"
+#include "dali/pipeline/graph/op_graph2.h"
+#include "dali/test/timing.h"
+
+namespace dali {
+
+namespace exec2 {
+namespace test {
```

Comment:
I think this deserves some tests that would check the CPU <-> GPU transitions (and Mixed?) - basically anything that works accross backends. There is a lot of code that does some things based on either the data not being GPU-placed or the operator not being CPU-backend.

I would probably suggest something that would introduce some significant work or sleeps in given operator, so we can test if the waits when going from one backend to another are doing their thing.

Reply:
On it.

Reply:
Done.